packages/payment-processor/src/payment/erc777-stream.ts

```diff
@@ -0,0 +1,55 @@
+import { ContractTransaction, Signer } from 'ethers';
+
+import { ClientTypes, ExtensionTypes, PaymentTypes } from '@requestnetwork/types';
+
+import {
+  getPaymentNetworkExtension,
+  getProvider,
+  getRequestPaymentValues,
+  validateRequest,
+} from './utils';
+import { Framework } from '@superfluid-finance/sdk-core';
+
+/**
+ * Processes a transaction to pay an ERC777 stream Request.
+ * @param request
+ * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
+ * @param _overrides optionally, override default transaction values, like gas.
+ */
+export async function payErc777StreamRequest(
+  request: ClientTypes.IRequestData,
+  signer: Signer,
+): Promise<ContractTransaction> {
+  const id = getPaymentNetworkExtension(request)?.id;
+  if (id !== ExtensionTypes.ID.PAYMENT_NETWORK_ERC777_STREAM) {
+    throw new Error('Not a supported ERC777 payment network request');
+  }
+  validateRequest(request, PaymentTypes.PAYMENT_NETWORK_ID.ERC777_STREAM);
+  const networkName =
+    request.currencyInfo.network === 'private' ? 'custom' : request.currencyInfo.network;
+  // console.log('networkName:', networkName);
+
+  const sf = await Framework.create({
+    networkName: networkName,
+    provider: signer.provider ? signer.provider : getProvider(),
+    dataMode: 'WEB3_ONLY',
+    resolverAddress: '0x8e4C131B37383E431B9cd0635D3cF9f3F628EDae', //this is how you get the resolver address
+    protocolReleaseVersion: 'test',
+  });
+  const superSigner = sf.createSigner({
+    signer: signer,
+    provider: signer.provider,
+  });
+  const superToken = await sf.loadSuperToken(request.currencyInfo.value);
+  // console.log('superToken.address:', superToken.address);
+  // TODO: use expectedStartDate to compute offset between start of invoicing and start of payment
```

Comment:
TODO here

Reply:
I've precised it in the description of this first version.

Reply:
I don't understand your message, but please remove TODOs from your code before merging. If that's something you want to improve later, rename it `FIXME`

Reply:
Explain it with an example or a "path" to visualize the spec that refer to it